hermes-management/src/main/java/pl/allegro/tech/hermes/management/api/WorkloadConstraintsEndpoint.java

```diff
@@ -0,0 +1,101 @@
+        this.service = service;
+    }
+
+    @GET
+    @Produces(APPLICATION_JSON)
+    @RolesAllowed(Roles.ANY)
+    @ApiOperation(value = "All workload constraints", response = List.class, httpMethod = HttpMethod.GET)
+    public ConsumersWorkloadConstraints getConsumersWorkloadConstraints() {
+        return service.getConsumersWorkloadConstraints();
+    }
+
+    @PUT
+    @Path("/topic")
+    @Consumes(APPLICATION_JSON)
+    @Produces(APPLICATION_JSON)
+    @RolesAllowed(Roles.ADMIN)
+    @ApiOperation(value = "Create or update topic constraints", response = String.class, httpMethod = HttpMethod.PUT)
+    public Response createOrUpdateTopicConstraints(@Valid TopicConstraints topicConstraints) {
+        if (service.constraintsExist(topicConstraints.getTopicName())) {
+            service.updateConstraints(topicConstraints.getTopicName(), topicConstraints.getConstraints());
+            return Response.status(OK).build();
+        } else {
+            service.createConstraints(topicConstraints.getTopicName(), topicConstraints.getConstraints());
+            return Response.status(CREATED).build();
+        }
+    }
+
+    @DELETE
+    @Path("/topic/{topicName}")
+    @RolesAllowed(Roles.ADMIN)
+    @ApiOperation(value = "Remove topic constraints", response = String.class, httpMethod = HttpMethod.DELETE)
+    public Response deleteTopicConstraints(@PathParam("topicName") String topicName) {
+        service.deleteConstraints(TopicName.fromQualifiedName(topicName));
+        return Response.status(OK).build();
+    }
+
+    @PUT
+    @Path("/subscription")
+    @Consumes(APPLICATION_JSON)
+    @Produces(APPLICATION_JSON)
+    @RolesAllowed(Roles.ADMIN)
+    @ApiOperation(value = "Create or update subscription constraints", response = String.class, httpMethod = HttpMethod.PUT)
+    public Response createOrUpdateSubscriptionConstraints(@Valid SubscriptionConstraints subscriptionConstraints) {
+        if (service.constraintsExist(subscriptionConstraints.getSubscriptionName())) {
+            service.updateConstraints(subscriptionConstraints.getSubscriptionName(), subscriptionConstraints.getConstraints());
+            return Response.status(OK).build();
+        } else {
+            service.createConstraints(subscriptionConstraints.getSubscriptionName(), subscriptionConstraints.getConstraints());
+            return Response.status(CREATED).build();
+        }
+    }
+
+    @DELETE
+    @Path("/subscription/{topicName}/{subscriptionName}")
+    @RolesAllowed(Roles.ADMIN)
+    @ApiOperation(value = "Remove subscription constraints", response = String.class, httpMethod = HttpMethod.DELETE)
+    public Response deleteSubscriptionConstraints(@PathParam("topicName") String topicName,
+                                                  @PathParam("subscriptionName") String subscriptionName) {
+        service.deleteConstraints(SubscriptionName.fromString(String.format("%s$%s", topicName, subscriptionName)));
```

Comment:
In my opinion it is not a right place to show internal representation of subscription name.
It will be better to do something like: `new SubscriptionName(subscriptionName, TopicName.fromQualifiedName(topicName))`